Create flake issue creator

dhiller opened the pull request:
This PR creates the tool `flake-issue-creator` that creates GitHub issues for the flakes discovered from the job test results. It uses the same data that the flakefinder uses for creating the reports and it uses the same algorithm to fetch and sort the flaky test data by severity.

It then works on the data in two passes:
pass 1: extract possible temporary cluster failures, create an issue or reopen existing issue and add comment for every lane that has a possible cluster failure
pass 2: create an issue or reopen existing issue and add comment for the remaining flaky tests

Following rules for reusing issues are implemented:
* tool does not add another comment on an issue that has been created or commented on less than 12h ago (in case the job is rerun within less than 12h this would trigger a lot of github mail spam). interval is configurable
* since the test names are very long there is a chance of not finding older issues (the query must not be more than 256 characters long). So if there's a `test_id` in the test title it is searched for test_id with priority

In a later PR we can add automatical assignment to sig if the test is tagged as proposed `[owner:@sig-compute]`

**Note to the reviewer:** the current job configuration executes with `dry-run=true` which just prints the actions that would have been done against github. I plan to leave that for a while, ask build watcher to verify if what the `flake-issue-creator` does matches the observation and then after a while remove the dry run. 

Closes #949

dhiller commented:
/cc @fgimenez

fgimenez reviewed (commented):
Very nice thanks! A couple of nits.

Looks like the error you are getting is because of a change in the images used for the rehearse jobs that happened in this PR https://github.com/kubevirt/project-infra/pull/1027 which I didn't notice, I'll prepare a PR to revert them

fgimenez commented:
> I'll prepare a PR to revert them

Proposed here https://github.com/kubevirt/project-infra/pull/1038

fgimenez commented on robots/cmd/flake-issue-creator/BUILD.bazel:
```diff
@@ -0,0 +1,66 @@
+    image = ":app",
+    registry = "quay.io",
+    repository = "kubevirtci/flake-issue-creator",
+    tag = "$(container_tag)",
```
This is how we use stamping to set the tag of the rehearse image now https://github.com/kubevirt/project-infra/blob/master/external-plugins/rehearse/plugin/BUILD.bazel#L42 with this in place the call to test looks like https://github.com/kubevirt/project-infra/blob/master/github/ci/prow-deploy/files/jobs/kubevirt/project-infra/project-infra-presubmits.yaml#L56 and the call to build and push https://github.com/kubevirt/project-infra/blob/master/github/ci/prow-deploy/files/jobs/kubevirt/project-infra/project-infra-postsubmits.yaml#L24

dhiller replied:
Thanks for pushing me into that direction :+1:

dhiller commented:
> Very nice thanks! A couple of nits.
> 
> Looks like the error you are getting is because of a change in the images used for the rehearse jobs that happened in this PR #1027 which I didn't notice, I'll prepare a PR to revert them

Thanks for the review, addressed your comments, PTAL.

fgimenez reviewed (approved):
/lgtm
/approve

nice thanks! sorry for missing this PR for so long
